global-control-plane.md

```diff
@@ -0,0 +1,284 @@
+
+```bash
+oc create route passthrough kong-kong-manager-tls --port=kong-manager-tls --service=kong-kong-manager -n kong
+oc create route passthrough kong-kong-admin-tls --port=kong-admin-tls --service=kong-kong-admin -n kong
+```
+
+- update the ADMIN URI in the deployment
+
+```bash
+oc patch deploy -n kong kong-kong -p "{\"spec\": { \"template\" : { \"spec\" : {\"containers\":[{\"name\":\"proxy\",\"env\": [{ \"name\" : \"KONG_ADMIN_API_URI\", \"value\": \"$(oc get route -n kong kong-kong-admin -ojsonpath='{.spec.host}')\" }]}]}}}}"
+```
+
+- save the cluster/clustertelemetry endpoints for later
+
+```bash
+export CLUSTER_URL=$(oc get svc kong-kong-cluster -ojson | jq -r '.status.loadBalancer.ingress[].hostname')
+export CLUSTER_TELEMETRY_URL=$(oc get svc kong-kong-clustertelemetry -ojson | jq -r '.status.loadBalancer.ingress[].hostname')
+```
+
+- check the management UI is working at:
+
+```bash
+oc get route kong-kong-manager --template='{{.spec.host}}'
+```
+
+- check the admin endpoint is available
+
+```bash
+http `oc get route kong-kong-admin --template='{{.spec.host}}'` | jq .version
+"2.8.1.1-enterprise-edition"
+```
+
+## Installing the Mesh in Openshift
+
+For the alternate installation on a VM refer to [Kong in a VM](./gateway-multizone/kong-vm.md#kong-in-a-vm)
+
+### Install Mesh Global Control Plane
+
+```bash
+kumactl install control-plane --cni-enabled --mode=global --license-path=./license.json | oc apply -f -
+```
+
+If running Kong Mesh in a multi-zone deployment, the file must be passed to the global kuma-cp. In this mode, Kong Mesh automatically synchronizes the license to the remote kuma-cp, therefore this operation is only required on the global kuma-cp.
+
+Enable TLS for the default mesh
+
+```bash
+kubectl apply -f mesh/mtls.yaml -n kong-mesh-system
+```
+
+Expose the svc
+
+```bash
+oc create route passthrough kong-mesh-control-plane --service=kong-mesh-control-plane --port=https-api-server 
+```
+
+Check it is working
+
+```bash
+$ http --verify=false https://`oc get route kong-mesh-control-plane -n kong-mesh-system -ojson | jq -r .spec.host`/config | jq .mode
+"global"
+```
+
+Create the Global Remote Sync route and extract the endpoint:
+
+```bash
+oc create route passthrough kong-mesh-global-zone-sync --service=kong-mesh-global-zone-sync --port=global-zone-sync -n kong-mesh-system
+export GLOBAL_SYNC=$(oc get route -n kong-mesh-system kong-mesh-global-zone-sync --template='{{ .spec.host }}'):443
+```
+
+### Install the Mesh Control Plane in Zone 1
+
+Login to the `kong-dp1` cluster
+
+```bash
+kumactl install control-plane --cni-enabled --mode=zone --zone=kumadp1 --ingress-enabled --kds-global-address grpcs://${GLOBAL_SYNC} | oc apply -f -
+```
+
+Check the zone is discovered in the global. Connect to the global control plane cluster to get the route
+
+```bash
+$ http `oc get route kong-mesh-control-plane -n kong-mesh-system -ojson | jq -r .spec.host`/status/zones                            
```

Comment:
https --verify=false `oc get route kong-mesh-control-plane -n kong-mesh-system -o json | jq -r .spec.host`/status/zones

Reply:
Yes I discovered the `https` command later haha, thanks